cli/command/container/diff_test.go

```
@@ -0,0 +1,95 @@
+package container
+
+import (
+	"context"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/docker/api/types/container"
+	"github.com/pkg/errors"
+	"gotest.tools/v3/assert"
+)
+
+func TestRunDiff(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		containerDiffFunc: func(
+			ctx context.Context,
+			containerID string,
+		) ([]container.FilesystemChange, error) {
+			return []container.FilesystemChange{
+				{
+					Kind: 0, // Modified ("C")
+					Path: "/path/to/file0",
+				},
+				{
+					Kind: 1, //  Added ("A")
+					Path: "/path/to/file1",
+				},
+				{
+					Kind: 2, // Deleted ("D")
+					Path: "/path/to/file2",
+				},
+			}, nil
+		},
+	})
+
+	cmd := NewDiffCommand(cli)
+	cmd.SetOut(io.Discard)
+
+	containerId := "a61d6b418006"
+	cmd.SetArgs([]string{containerId})
+
+	err := cmd.Execute()
+	assert.Assert(t, err == nil)
+
+	diff := strings.SplitN(cli.OutBuffer().String(), "\n", 3)
+	file0 := strings.TrimSpace(diff[0])
+	file1 := strings.TrimSpace(diff[1])
+	file2 := strings.TrimSpace(diff[2])
+
+	assert.Assert(t, len(diff) == 3)
+	assert.Assert(t, file0 == "C /path/to/file0")
+	assert.Assert(t, file1 == "A /path/to/file1")
+	assert.Assert(t, file2 == "D /path/to/file2")
+}
+
+func TestRunDiffClientError(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		containerDiffFunc: func(
+			ctx context.Context,
+			containerID string,
+		) ([]container.FilesystemChange, error) {
+			return nil, errors.New("client error")
+		},
+	})
+
+	cmd := NewDiffCommand(cli)
+	cmd.SetOut(io.Discard)
```

Can you also discard the error output here? 

```go
cmd.SetOut(io.Discard)
cmd.SetErr(io.Discard)
```

It prevents the error that's part of the test from being printed, which can make it confusing when reading test logs;

```console
=== RUN   TestRunDiffClientError
Error: client error
--- PASS: TestRunDiffClientError (0.00s)
PASS
```

Suppressing the output still makes the test pass, but doesn't print the output;

```
=== RUN   TestRunDiffClientError
--- PASS: TestRunDiffClientError (0.00s)
PASS
```